src/org/daisy/dotify/formatter/impl/core/TableOfContentsImpl.java

```diff
@@ -73,54 +82,122 @@ public void endBlock() {
 
 	@Override
 	public Block newBlock(String blockId, RowDataProperties rdp) {
-		Block b = super.newBlock(blockId, rdp);
+        Block b = super.newBlock(blockId, rdp);
 		if (!currentAncestorTocBlocks.isEmpty()) {
-			tocBlockForBlock.put(b, currentAncestorTocBlocks.peek());
+            tocBlockForBlock.put(b, currentAncestorTocBlocks.peek());
 		}
 		return b;
 	}
 
 	@Override
 	public void startEntry(String refId) {
-		if (inTocEntry) {
-			throw new RuntimeException("toc-entry may not be nested");
```

Comment:
I think this check was nice to have. And `startEntryOnResumed()` could do the same check.

Reply:
Done (2x).